src/neo/SmartContract/Native/Tokens/NeoToken.Economic.cs

```diff
@@ -0,0 +1,178 @@
+            StorageItem item = engine.Snapshot.Storages.GetAndChange(CreateStorageKey(Prefix_GasPerBlock));
+            item.Value = gasPerBlock.ToByteArray();
+            return true;
+        }
+
+        [ContractMethod(0_05000000, CallFlags.AllowModifyStates)]
+        private bool SetNeoHoldersRewardRatio(ApplicationEngine engine, int neoHoldersRewardRatio)
+        {
+            if (neoHoldersRewardRatio < 0 || neoHoldersRewardRatio > byte.MaxValue) return false;
+            if (!CheckCommitteeWitness(engine)) return false;
+            StorageItem item = engine.Snapshot.Storages.GetAndChange(CreateStorageKey(Prefix_NeoHoldersRewardRatio));
+            item.Value = new byte[] { (byte)neoHoldersRewardRatio };
+            return true;
+        }
+
+        [ContractMethod(0_05000000, CallFlags.AllowModifyStates)]
+        private bool SetCommitteeRewardRatio(ApplicationEngine engine, int committeesRewardRatio)
+        {
+            if (committeesRewardRatio < 0 || committeesRewardRatio > byte.MaxValue) return false;
+            if (!CheckCommitteeWitness(engine)) return false;
+            StorageItem item = engine.Snapshot.Storages.GetAndChange(CreateStorageKey(Prefix_CommitteeRewardRatio));
+            item.Value = new byte[] { (byte)committeesRewardRatio };
+            return true;
+        }
+
+        [ContractMethod(0_05000000, CallFlags.AllowModifyStates)]
+        private bool SetVotersRewardRatio(ApplicationEngine engine, int votersRewardRatio)
+        {
+            if (votersRewardRatio < 0 || votersRewardRatio > byte.MaxValue) return false;
+            if (!CheckCommitteeWitness(engine)) return false;
+            StorageItem item = engine.Snapshot.Storages.GetAndChange(CreateStorageKey(Prefix_VotersRewardRatio));
+            item.Value = new byte[] { (byte)votersRewardRatio };
+            return true;
+        }
+
+        [ContractMethod(1_00000000, CallFlags.AllowStates)]
+        public BigInteger GetGasPerBlock(StoreView snapshot)
+        {
+            return new BigInteger(snapshot.Storages.TryGet(CreateStorageKey(Prefix_GasPerBlock)).Value);
+        }
+
+        [ContractMethod(1_00000000, CallFlags.AllowStates)]
+        public int GetNeoHoldersRewardRatio(StoreView snapshot)
+        {
+            return snapshot.Storages.TryGet(CreateStorageKey(Prefix_NeoHoldersRewardRatio)).Value[0];
+        }
+
+        [ContractMethod(1_00000000, CallFlags.AllowStates)]
+        public int GetCommitteeRewardRatio(StoreView snapshot)
+        {
+            return snapshot.Storages.TryGet(CreateStorageKey(Prefix_CommitteeRewardRatio)).Value[0];
+        }
+
+        [ContractMethod(1_00000000, CallFlags.AllowStates)]
+        public int GetVotersRewardRatio(StoreView snapshot)
+        {
+            return snapshot.Storages.TryGet(CreateStorageKey(Prefix_VotersRewardRatio)).Value[0];
+        }
+
+        private int GetTotalRewardRatio(StoreView snapshot)
```

Comment:
Maybe we should ensure that the sum it's 100 and skip this method?

Reply:
I can merge `SetVotersRewardRatio`, `SetCommitteeRewardRatio` and `SetNeoHoldersRewardRatio` into one function.